types/sharedb/index.d.ts

```diff
@@ -31,6 +31,13 @@ declare class sharedb extends EventEmitter {
     extraDbs: {[extraDbName: string]: sharedb.ExtraDB};
     milestoneDb?: sharedb.MilestoneDB;
 
+    readonly projections: Readonly<{
+        [name: string]: {
+            readonly target: string;
+            readonly fields: Readonly<ProjectionFields>;
+        };
```

Comment:
Do you think it would be nice to extract this as a `Projection` interface in case people need to reference it?

```suggestion
        [name: string]: Projection;
```

Reply:
Shouldn't we update the existing `Projection` interface to have `readonly` properties, then? Surely we'd want this type safety everywhere?

Reply:
I had been afraid of breakages for people referencing the existing `Projection` type. On second look, it is not exported, so modifying it internally is fine.

The existing type was used in `SubmitRequest`, so there could still potentially be a breakage if someone was doing something weird like `submitRequest.projection.target = 'foo'`. That's very very unlikely, so I think it'll be fine.

Reply:
Also people shouldn't be doing that, right? That's the whole point of this change, so if anything it's a bugfix. I think we're fine.